### ISSUE TITLE
Emit OSError on bad read()

### ISSUE THREAD

**Issue by effigies** [animalize/pyzstd#20]
The 0.15.4 release changed the exception raised by the following test code from `EOFError` to `ZstdError`:

```Python
import io
import pyzstd

pyzstd.ZstdFile(io.BytesIO(b'bad'), mode='rb').read()
```

The reason this is problematic for me is that I have `pyzstd` as an optional dependency, so I'm having to move from a list of builtin exception types to conditionally available types. It would probably end up looking something like this:

```Python
exceptions = (EOFError, OSError)
try:
    import pyzstd
    exceptions += (pyzstd.ZstdError, )
except:
    pass

def read_and_do_something(obj):
    try:
        res = obj.read()
    except exceptions as e:
        ...
```

In comparison, `gzip.GzipFile` and `bz2.BZ2File` both raise `OSError`s:

```Python
import bz2
import gzip
import io
import pytest

with pytest.raises(OSError):
    gzip.GzipFile(fileobj=io.BytesIO(b'bad'), mode='rb').read()
with pytest.raises(OSError):
    bz2.BZ2File(io.BytesIO(b'bad'), mode='rb').read()
```

In the case of bz2, it's directly, while gzip raises a `BadGzipFile`, which subclasses `OSError`. I wonder if it would work for you to derive `ZstdError` from `OSError`, like `BadGzipFile`, or possibly `ZstdError` in your IO functions and raising `OSError` from them?

**Comment by animalize** [comment]
TLDR: Won't fix, very thing is in the right way. Please use conditionally available types.

This issue is exposed by this `zstd` v1.5.4 change: https://github.com/facebook/zstd/pull/3175
Before the change, `zstd` waited for at least 5 bytes to start checking Magic Number (the first 4 bytes). Now it checks when the first byte arrives.

Back to Python modules, your test code on CPython 3.10:
```
class      exception    super class
GzipFile   BadGzipFile  OSError
BZ2File    OSError      Exception
LZMAFile   LZMAError    Exception
ZstdFile   ZstdError    Exception
```

[`OSError document`](https://docs.python.org/3/library/exceptions.html#OSError) says:
> This exception is raised when a system function returns a system-related error, including I/O failures such as “file not found” or “disk full” (not for illegal argument types or other incidental errors).

So literally, `OSError` is not pertinent. It's not a system-related problem. 

`bz2` doesn't have its own error type. It's also inappropriately raises `OSError` for data error: 
https://github.com/python/cpython/blob/v3.10.10/Modules/_bz2module.c#L147-L150

Considering the year of adding these modules, using `OSError` may be an early immature API:
`zlib` module was added in 1997
`bz2` module was added in 2002
`lzma` module was added in 2011  (`pyzstd`'s behavior is consistent with this module)

**Comment by effigies** [comment]
Got it. Thanks for the response.
